hibernate-core/src/test/java/org/hibernate/test/bytecode/enhancement/lazy/LazyCollectionLoadingTest.java

```diff
@@ -101,7 +102,40 @@ public void testTransaction() {
             checkDirtyTracking( parent );
 
             assertThat( children1, sameInstance( children2 ) );
+
+            assertFalse( isInitialized( children1 ) );
+            assertThat( children1.size(), equalTo( CHILDREN_SIZE ) );
+            assertTrue( isInitialized( children1 ) );
+        } );
+    }
+
+    @Test
+    @TestForIssue( jiraKey = "HHH-14620" )
+    public void testTransaction_noProxy() {
+        doInHibernate( this::sessionFactory, s -> {
+            // find will not return a proxy, which is exactly what we want here.
+            Parent parent = s.find( Parent.class, parentID );
+            assertThat( parent, notNullValue() );
+            assertThat( parent, not( instanceOf( HibernateProxy.class ) ) );
+            assertThat( parent, not( instanceOf( HibernateProxy.class ) ) );
```

Comment:
I know this might be copied from existing method, but is it intentional that we repeat the same assertion twice (or do we rely on some side effect)?

Reply:
Fixed, thanks